src/main/java/com/fundynamic/d2tm/game/map/Map.java

```diff
@@ -106,7 +110,7 @@ public Terrain getTerrain() {
 
         private TerrainFacing determineFacing() {
             Terrain terrain = getTerrain();
-            FacingDeterminer facingDeterminer = new FacingDeterminer();
+            TerrainFacingDeterminer facingDeterminer = new TerrainFacingDeterminer();
```

Comment:
I almost feel a factory approach coming here: `factory.getFacingDeterminer(type)` etc. So we can abstract it. Renaming is nice (now) btw, so it is very explicit for terrain! :)